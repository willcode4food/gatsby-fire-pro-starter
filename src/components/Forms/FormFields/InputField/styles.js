import styled from '@emotion/styled'
import { COLORS } from 'utils/styleHelpers'

export const InputFieldStyle = styled.input`
	color: ${COLORS.DARK_GRAY};
	font-size: inherit;
	line-height: inherit;
	width: 100%;
	box-shadow: rgb(238, 238, 238) 0px 0px 0px 1px inset;
	padding-left: 10px;
	padding-right: 4px;
	padding-top: 8px;
	padding-bottom: 8px;
	background: ${COLORS.WHITE};
	font-family: inherit;
	display: inline-block;
	vertical-align: middle;
	-webkit-appearance: none;
	border-radius: 4px;
	margin: 0px;
	border-width: 0px;
	border-style: initial;
	border-color: initial;
	border-image: initial;
	transition: background 0.5s, color 0.5s;
	.dark & {
		background: ${COLORS.BLACK};
		color ${COLORS.WHITE};
	}
`
