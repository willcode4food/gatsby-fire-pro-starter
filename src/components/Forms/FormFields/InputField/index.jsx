import React from 'react'
import PropTypes from 'prop-types'
import { InputFieldStyle } from './styles'

function InputField({ register: registerProp, name, ...props }) {
    const { register, ...registerValues } = registerProp
    return <InputFieldStyle {...register(`${name}`, { ...registerValues })} {...props} />
}

InputField.propTypes = {
    input: PropTypes.any,
    meta: PropTypes.any,
    registerProp: PropTypes.func,
}

export default InputField
