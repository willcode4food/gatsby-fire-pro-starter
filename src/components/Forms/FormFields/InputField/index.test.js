/* eslint-disable no-undef */
import React from 'react'
import renderer from 'react-test-renderer'
import serializer from 'jest-emotion'
import InputField from './index.jsx'

expect.addSnapshotSerializer(serializer)

describe('InputField', () => {
	it('renders correctly', () => {
		const tree = renderer
			.create(<InputField name="lastName" placeholder="Last Name" type="text" aria-label="Last Name" />)
			.toJSON()
        console.log("tree", tree)
			
		expect(tree).toMatchSnapshot()
	})
})
