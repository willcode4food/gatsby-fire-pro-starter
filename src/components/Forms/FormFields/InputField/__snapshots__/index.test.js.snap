// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField renders correctly 1`] = `
.emotion-0 {
  font-size: inherit;
  line-height: inherit;
  width: 100%;
  box-shadow: #C4C8D8 0px 0px 0px 1px inset;
  padding-left: 10px;
  padding-right: 4px;
  padding-top: 13px;
  padding-bottom: 13px;
  background: #FFF;
  box-sizing: border-box;
  overflow-wrap: break-word;
  overflow-x: visible;
  overflow-y: visible;
  -webkit-text-size-adjust: 100%;
  text-size-adjust: 100%;
  font-family: inherit;
  display: inline-block;
  vertical-align: middle;
  -webkit-appearance: none;
  border-radius: 4px;
  margin: 0px;
  border-width: 0px;
  border-style: initial;
  border-color: initial;
  border-image: initial;
  -webkit-transition: background 0.5s,color 0.5s;
  transition: background 0.5s,color 0.5s;
}

.dark .emotion-0 {
  background: #263238;
  color: #FFF;
}

<input
  aria-label="Last Name"
  className="emotion-0"
  name="lastName"
  placeholder="Last Name"
  type="text"
/>
`;
